src/hooks/useConnectionStatus.ts

```diff
@@ -0,0 +1,19 @@
+import { useState, useEffect } from "react"
+import NetInfo from "@react-native-community/netinfo"
+
+export const useConnectionStatus = (): boolean => {
+  const [isConnected, setIsConnected] = useState<boolean>(true)
+
+  useEffect(() => {
+    const unsubscribe = NetInfo.addEventListener((state) => {
+      // netInfo state comes as null while unresolved so to avoid flicker we only set component state
+      // if the netInfo state is resolved to boolean
+      if (typeof state.isInternetReachable === "boolean") {
```

Comment:
I feel like this type check could be avoided if we do something like:

```typescript
useEffect(() => {
  const unsubscribe = NetInfo.addEventListener((netInfo) => {
    setIsConnected(Boolean(netInfo?.isInternetReachable))
  })
  return unsubscribe
}, [])
```
Could that work? I feel like it is less hard to read the `Boolean` conversion than the type check.

Reply:
Unfortunately, this won't work. `undefined` we get with optional chaining will convert to `false` and we will get flicker we were getting before we used this check here.